### PR TITLE
Bring the CONTRIBUTING.md developer documentation up to date

### PR DESCRIPTION
**Pull request by adamretter**
Updates the CONTRIBUTING.md developer documentation so that it is up to date and also adds a copy of [Community Code Review and Merge Policy for the exist-db/exist Git Repository v2](https://docs.google.com/document/d/1O724HZ2s3ffhclMATVpVTIskdCNhazA11A7GqxlGe24) as Markdown to the Git repo so that it can be easily referenced.

### CODE REVIEW

File: CONTRIBUTING.md
Comment (joewiz) [inline CONTRIBUTING.md:24]: I think this example should be `hotfix` rather than `feature`, no?

File: CONTRIBUTING.md
Comment (joewiz) [inline CONTRIBUTING.md:89]: s/one of more/one or more (I think?)

File: CONTRIBUTING.md
Comment (joewiz) [inline CONTRIBUTING.md:91]: s/Under these circumstance pull-request/In this circumstance a pull request

File: CONTRIBUTING.md
Comment (joewiz) [inline CONTRIBUTING.md:3]: s/Slack Channel/Slack workspace

File: CONTRIBUTING.md
Comment (joewiz) [inline CONTRIBUTING.md:5]: s/Pull-Requests/pull requests

File: CONTRIBUTING.md
Comment (joewiz) [inline CONTRIBUTING.md:64]: s/Pull Requests/pull requests

File: CONTRIBUTING.md
Comment (joewiz) [inline CONTRIBUTING.md:91]: s/pull-request/pull request

File: CONTRIBUTING.md
Comment (joewiz) [inline CONTRIBUTING.md:130]: Does anyone really use the tool linked here? If so, then fine. If not, we could delete this section.

File: CONTRIBUTING.md
Comment (joewiz) [inline CONTRIBUTING.md:143]: This point should perhaps be moved above to the "Pull Requests and Code Review" section. It's true regardless of whether someone uses the "AVH Edition of GitFlow tools."

File: CONTRIBUTING.md
Comment (joewiz) [reply CONTRIBUTING.md:5]: s/, therefore/, so (or change "," to ";")

File: CONTRIBUTING.md
Comment (joewiz) [inline CONTRIBUTING.md:8]: s/ like/-like

File: CONTRIBUTING.md
Comment (adamretter) [reply CONTRIBUTING.md]: Thanks @joewiz I pushed a commit to address this.

File: CONTRIBUTING.md
Comment (adamretter) [reply CONTRIBUTING.md]: Thanks @joewiz I pushed a commit to address this.

File: CONTRIBUTING.md
Comment (adamretter) [reply CONTRIBUTING.md]: Thanks @joewiz I pushed a commit to address this.

File: CONTRIBUTING.md
Comment (adamretter) [reply CONTRIBUTING.md]: Thanks @joewiz I pushed a commit to address this.

File: CONTRIBUTING.md
Comment (adamretter) [reply CONTRIBUTING.md:130]: Yes they do.

File: CONTRIBUTING.md
Comment (adamretter) [reply CONTRIBUTING.md]: Thanks @joewiz I pushed a commit to address this.

File: CONTRIBUTING.md
Comment (adamretter) [reply CONTRIBUTING.md:143]: I am not sure that would work as this is specifically about how to use AVH GitFlow.

**Comment by adamretter** [comment]
> We ran out of time today, so we're going to discuss this at next week's community call.

Interesting. Is there something contentious about this which means it needs special discussion as opposed to just going through the standard PR and review process?

**Review by reinhapa — COMMENTED** [review]
@adamretter would be cool to add those spaces issues that where mentioned from Codacy...

**Comment by adamretter** [comment]
@reinhapa I think Codacy may be wrong there. The spec for lists in GFM is here - https://github.github.com/gfm/#list-items

**Comment by adamretter** [comment]
> ok, could we configure Codacy in a way to use this spec or ignore those file then?

@reinhapa Done!
Screenshot 2023-05-25 at 13 59 29

**Review by reinhapa — APPROVED** [review]

File: CONTRIBUTING.md
Comment (line-o) [reply CONTRIBUTING.md:143]: Is AVH GitFlow used by anyone contributing to this project?

**Comment by wolfgangmm** [comment]
The concerns raised at last Monday's (June 5) community meetup were mainly connected to the remediation process, which in its current form does not seem to work well, i.e. the goals we discussed a while ago: less stale PRs, quicker responses etc., have not been realized. We should discuss how this could be adjusted to achieve the desired effect. Last Monday not enough people attended, so the discussion has to be continued and all input will be welcome.

**Comment by adamretter** [comment]
@wolfgangmm So presumably that would lead to a version 3 of the `Community Code Review and Merge Policy? As this PR has the current version (i.e. 2), I think this PR could be merged now, and later if a v3 emerges then a new PR could be sent to update this.

Additionally, if we are discussing a new policy, as it involves the core developers (of which I am one), please make me aware of when these discussions will happen so that I might have the option of contributing.

**Comment by duncdrum** [comment]
@adamretter the community call notes have this info. They are freely accessible

**Comment by adamretter** [comment]
@duncdrum I read the Community Call minutes from the `2023-06-05` meeting that were published here - https://docs.google.com/document/d/16oXbhBiR2R9gJXZV6jCAhQO1Y-DjYFfQqaxFSc9g_YU/

There is nothing in the minutes about a vote. The minutes in fact state that it was proposed to accept this PR. Additionally the minutes state that some thought is being given to writing an updated Contributor Guide (presumably a v3).

So I still don't see anything that prohibits this PR from being merged immediately (as it just represents the situation as it is today). Waiting for a revised version that may take a long time to produce or may not even happen seems counter productive.

There is also no information in the Community Call minutes about when revisions to a Contributor Guide will be discussed in future. I would like forewarning of when those specific meetings will be between the core developers.

**Comment by duncdrum** [comment]
The minutes do not. JW put his vote to merge in the agenda. He wasn’t on the call, and that wasn’t the position reached after the discussion.

**Comment by adamretter** [comment]
Understood. I have no problem with the community process. However I would like to know the details. If it is not documented, then I have no basis to evaluate. So... questions...

1. This represents the true and community agreed approach at this time. Therefore it seems reasonable to merge it at this time... Do you agree?

2. If there are further discussions to be had about revising the policy. I have expressed that I would like to be involved. The minutes of the recent meeting give no hints as to when, where, and who might be involved in such discussions.  Or did I miss something from the minutes?

**Comment by duncdrum** [comment]
@adamretter the discussion was tabled two weeks ago at the community call and is on the call notes. Due to holidays the next call was actually two weeks later, where we discussed this. We decided to continue at the next call.

**Comment by adamretter** [comment]
Considering that attendance to the community call is arbitrary and different each week... For an important topic such as this, where there will need to be agreement from a quorum of core developers to impact any change. It seems to me that this needs to be scheduled and arranged with the core developers explicitly invited.

It's fine for the community to have ideas and sketch them out on paper, and I certainly welcome that. However, unless those people actually doing the work, that is, contributing code frequently to the eXist-db codebase (i.e. the core developers as recorded in the V2 of the community contributing guidelines) agree to it, then this will not succeed.

The community will have to, actively rather than passively, solicit support from the core developers, otherwise this is nothing but a thought exercise.

**Comment by adamretter** [comment]
I also still don't see why any part of these community discussions hold up a PR that truthfully reflects the agreements that are in place today

**Comment by windauer** [comment]
@adamretter and thats exactly what is happening here and what @wolfgangmm wrote in [his comment](https://github.com/eXist-db/exist/pull/4933#issuecomment-1580393826). In the community meetup notes from Monday which you linked it says: "WM, DP, JL, LW, DW voiced the opinion the the process laid out in the document did not work out as intended..." The people in the call discussed this topic ~50 min and came to the conclusion, that we 

a) don't want to merge this PR in its current form and 
b) we'd like to invite all eXist-db contributors to take part into that discussion, also the core devs. 

And thats the reason why we did not simply "decide" to not merge the PR but agreed to go in the discussion with everyone who is interested.

**Comment by duncdrum** [comment]
@adamretter you last two posts demonstrate the problematic aspects of status quo quite effectively.

At both calls (announcing the discussion 2 weeks ago, and at the discussion) three Java devs were present. So I m not sure what you expect. Nothing passively soliciting about it.

**Comment by adamretter** [comment]
@duncdrum I am not against these discussions. My observations and suggestions are meant to be helpful in this matter.

There was no announcement outside of the call. I would have hoped for something to have been posted in the Slack or sent out by email about when these discussions will happen. Perhaps that should be done?

Also 3 out of 6 does not represent a quorum. So I would think that you might want to find a time and date that works for all or a quorum of the core developers.

**Comment by adamretter** [comment]
@windauer the discussion and decisions are not documented in the minutes at all.

Wanting to change things is great, but it will take time!

My point is that this PR reflects what we "the community" have agreed in the past and documented. It reflects the rules as they stand today. Therefore there is nothing wrong with this PR.

By not merging this PR you are effectively invalidating and overriding a decision reached and agreed by the community. If such decisions by the community don't have any weight, then I suggest we simply abandon the community provcess and all go and do something else more productive!

**Comment by adamretter** [comment]
> @adamretter your last two posts demonstrate the problematic aspects of the status quo quite effectively.
>

@duncdrum I am not sure what is meant by this. The minutes don't help as there is almost nothing written there. Can you explain what you mean please?

**Comment by adamretter** [comment]
As this has been open for 6 months, I have now rebased it.
The documentation in this PR is an accurate reproduction of the standards and practices agreed by the community and documented by the community in various Community Group calls.

If individuals wish to challenge the current standards with a view to revising them, then I have no problem with that, and I would be happy to participate in such discussions, but this PR is not the place to do that. 

@dizzzz @reinhapa I would like to get this merged please?

**Review by dizzzz — APPROVED** [review]

**Comment by dizzzz** [comment]
nice readable text, good basis; we'll see how it works out.

**Comment by adamretter** [comment]
Thanks you @dizzzz :-)

**Pull request merged**
